test/lib/bamboo/adapters/test_adapter_test.exs

```diff
@@ -172,6 +172,38 @@ defmodule Bamboo.TestAdapterTest do
     end
   end
 
+  test "refute_email_delivered_with when email does not match" do
+    new_email(
+      to: {nil, "foo@bar.com"},
+      from: {nil, "foo@bar.com"},
+      text_body: "I really like coffee"
+    )
+    |> TestMailer.deliver_now()
+
+    refute_email_delivered_with(text_body: ~r/tea/)
+    refute_email_delivered_with(to: "bla")
```

Comment:
I think there can only be one assertion of this type per email delivery -- the received email params are only pulled for the first assertion. If you switch this line to `refute_email_delivered_with(to: [nil: "foo@bar.com"])` it passes which it should not? So I think for these additional assertions to be relevant, we need to call `deliver_now` again before each - similar to https://github.com/thoughtbot/bamboo/blob/3f79b73a4e27d9b27694fb4a72473caab46c857f/test/lib/bamboo/adapters/test_adapter_test.exs#L56-L63

Reply:
You're right, of course, will fix. 

I wonder though if  this is the best way to implement these assertions. It would be better to store the emails in an agent and match over all of them.  That would require a big refactor, but it might be worth it.

Reply:
1deb19c

Reply:
> I wonder though if this is the best way to implement these assertions. It would be better to store the emails in an agent and match over all of them. That would require a big refactor, but it might be worth it.

What do you think about keeping it as is for now and merging this and opening a new issue for the potential refactor?